File: 백준/Silver/2559. 수열/수열.py
import sys
input = sys.stdin.readline

n, k=map(int, input().split())
arr=list(map(int, input().split()))
sum_=[0]

for i in range(n):
    sum_.append(sum_[-1]+arr[i])


answer = -float("inf")
for i in range(k,n+1):
    num = sum_[i]-sum_[i-k]
    answer = answer if answer>=num else num
print(answer)

File: 백준/Silver/2559. 수열/수열.js
const readline = require("readline")
const rl = readline.createInterface({
input:process.stdin,
output:process.stdout

})

const lines = []

rl.on(("line"),(line)=>{
lines.push(line.trim())


}).on(("close"),()=>{
const [N,K] = lines[0].split(" ").map(Number)
const arr = lines[1].split(" ").map(Number)
let sum_ = arr.slice(0,K).reduce((acc, curr)=>(
    acc+=curr
),0)
let answer = sum_
let start = 0
let end = K-1
while(end+1<arr.length){
    end+=1
    sum_ = sum_-arr[start]+arr[end]
    //console.log("빼기:" ,arr[start], "더하기: ",arr[end],"총합", sum_)
    start+=1
    answer = Math.max(answer, sum_)
    
}
console.log(answer)
})